http: Unify shutdown/cleanup logic

Lazin opened the pull request:
The http client has a bunch of method related to shutdown. This PR attempts to unify all of them and leave only `stop` and `shutdown`. The `stop` is used to dispose the client and `shutdown` is used to disconnect.
This is just a refactoring. The logic is not changed.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
* none

Lazin committed:
http: Merge stop and shutdown_and_stop

The last method is just an alias for 'stop' added in order to use with
google mock. It's safe to just make the 'stop' method virtual and mock
it instead.

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

Lazin committed:
http: Remove 'fail_outstanding_futures'

The method is used by the 'base_transport' to invoke some logic in the
'stop' method. It's not necessary for the 'http' client because the
client can invoke 'shutdown' directly.

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

Lazin committed:
http: Add doc-comment for the shutdown/stop

..methods

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

oleiman commented on src/v/http/client.h:
```diff
@@ -116,7 +116,13 @@ class client
+    /// The 'shutdown' method can be used to disconnect the http client
+    /// from the server. The client can be reconnected after that.
+    /// The client can't be safely disposed after calling 'shutdown'. The 'stop'
+    /// method should be called and awaited first.
```
nitpick: slightly confusing phrasing. feel free to ignore.

```suggestion
    /// To permanently spin down a client for destruction, call the 'stop' method 
    /// instead.
```

idk something like that

Lazin replied:
done

oleiman reviewed (dismissed):
lgtm

rockwotj commented on src/v/http/client.h:
```diff
@@ -116,7 +116,13 @@ class client
+    /// from the server. The client can be reconnected after that.
+    /// To permanently spin down a client for destruction, call the 'stop'
+    /// method instead.
     using net::base_transport::shutdown;
```
🙏 thank you for this!

oleiman reviewed (dismissed):
🙏

Lazin committed:
http: Merge shutdown and shutdown_now

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

oleiman reviewed (dismissed):
Precise details are slipping my mind, but I recall an hypothesis that we could knock `client::_shutdown_now` off in favor of an equivalent flag internal to `base_transport`. What happened there?

Not a complaint - this version feels more like a straight refactor, which is cool w/ me.

Lazin commented:
> Precise details are slipping my mind, but I recall an hypothesis that we could knock `client::_shutdown_now` off in favor of an equivalent flag internal to `base_transport`. What happened there?

the method that resets the flag has unwanted side effects so I decided that it's simpler this way

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/67331-->
<!--ac4a1dfc-ee2b-49f6-8a2d-ffe67233f2f4-->
<!--7a587ddd-df63-4606-8fba-2378f3a57146-->
<!--2cf962fe-577c-41a2-b7bf-62da696adbb7-->
<!--80b77606-93f9-4654-bf0c-91c939ebce06-->
<!--title-->
## Retry command for [Build#67331](https://buildkite.com/redpanda/redpanda/builds/67331)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init
tests/rptest/tests/random_node_operations_test.py::RandomNodeOperationsTest.test_node_operations@{"cloud_storage_type":1,"compaction_mode":"sliding_window","enable_failures":true,"mixed_versions":false,"with_iceberg":true}
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_invalidate_keys
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--2fc75a41-1137-4e8c-99d8-a0bcdf50755f-->
<!--TEST RESULTS-->
<!--38ca1867-2871-4484-bf81-4e51879f45c0-->
<!--TEST RESULTS-->
<!--b2436444-ca4f-43bd-8a04-ddec8f216370-->
<!--TEST RESULTS-->
<!--adfe624e-3228-47c8-986d-7e7ef6f9d0d6-->
<!--TEST RESULTS-->
<!--a7586cd9-d69c-4c81-9c96-63ae6819406d-->
<!--TEST RESULTS-->
<!--6aa4f9e4-8d66-47d6-8115-e71cf0112845-->
<!--TEST RESULTS-->
<!--c406c784-41ae-4c2b-825d-382c3db3c0d9-->
<!--TEST RESULTS-->
<!--71d7368a-b27a-4716-aaca-6fa0b92a811e-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67331">build#67331</a></summary>


| test_class                                   | test_method                                       | test_arguments                                                                                                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:---------------------------------------------|:--------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| MasterTestSuite                              | test_new_segment_upload_fuzz                      |                                                                                                                                        | unit        | https://buildkite.com/redpanda/redpanda/builds/67331#019769ed-ebbe-471c-84a7-548130b85571 | FAIL          | 0/1      |                                                                                                                                                                         |
| CloudStorageTimingStressTest                 | test_cloud_storage                                | {"cleanup_policy": "compact,delete"}                                                                                                   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67331#01976a4e-5ecd-44b1-ad2b-e115fd0f41dc | FLAKY         | 18/21    | upstream reliability is '97.44525547445255'. current run reliability is '85.71428571428571'. drift is 11.73097 and the allowed drift is set to 50. The test should PASS |
| PartitionReassignmentsTest                   | test_add_partitions_with_inprogress_reassignments |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67331#01976a4e-5ed0-45fe-b7f7-7b8c8f122ec2 | FLAKY         | 19/21    | upstream reliability is '92.15384615384615'. current run reliability is '90.47619047619048'. drift is 1.67766 and the allowed drift is set to 50. The test should PASS  |
| RandomNodeOperationsTest                     | test_node_operations                              | {"cloud_storage_type": 1, "compaction_mode": "sliding_window", "enable_failures": true, "mixed_versions": false, "with_iceberg": true} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67331#01976a4e-5ecf-4a6c-9b1f-9618b55994cc | FAIL          | 0/1      | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest                          | test_admin_invalidate_keys                        |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67331#01976a4e-5ecf-4a6c-9b1f-9618b55994cc | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest                          | test_admin_revoke                                 |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67331#01976a4e-5ed0-45fe-b7f7-7b8c8f122ec2 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest                          | test_admin_whoami                                 |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67331#01976a4e-5ecd-44b1-ad2b-e115fd0f41dc | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest                          | test_init                                         |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67331#01976a4e-5ece-4605-9624-b804b6aff550 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| src/v/crypto/tests/crypto_bench_rpbench_test | src/v/crypto/tests/crypto_bench_rpbench_test      |                                                                                                                                        | unit        | https://buildkite.com/redpanda/redpanda/builds/67331#01976aa8-a724-404d-a46b-a0f1757e18d5 | FAIL          | 0/1      |                                                                                                                                                                         |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67380">build#67380</a></summary>


| test_class          | test_method                                            | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:--------------------|:-------------------------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RedpandaOIDCTlsTest | test_admin_invalidate_keys                             |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67380#0197780b-767f-4069-91bf-0dd103b81a6d | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest | test_admin_invalidate_keys                             |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67380#0197783b-8fb6-43e9-a335-00bedd4a586c | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest | test_admin_revoke                                      |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67380#0197780b-767f-46d7-a353-f47e6cea17ba | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest | test_admin_revoke                                      |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67380#0197783b-8fb6-4c1f-b211-85e4fd5b1e3f | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest | test_admin_whoami                                      |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67380#0197780b-7680-48ce-a751-823efed69e74 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest | test_admin_whoami                                      |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67380#0197783b-8fb4-45b1-b2d1-f864686ece88 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest | test_init                                              |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67380#0197780b-7680-43ed-8a99-23b919cb9983 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest | test_init                                              |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67380#0197783b-8fb5-4777-a16e-19b9d21024ff | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| TxUpgradeTest       | upgrade_does_not_change_tx_coordinator_assignment_test |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67380#0197783b-8fb6-4c1f-b211-85e4fd5b1e3f | FLAKY         | 16/21    | upstream reliability is '92.55813953488372'. current run reliability is '76.19047619047619'. drift is 16.36766 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67396">build#67396</a></summary>


| test_class                                        | test_method                                       | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 |
|:--------------------------------------------------|:--------------------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|
| RedpandaOIDCTlsTest                               | test_admin_invalidate_keys                        |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67396#019779bc-de18-4cf7-9e69-8532ccb2ddf1 | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest                               | test_admin_revoke                                 |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67396#019779bc-de19-4627-9331-1c59c1c0bfa5 | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest                               | test_admin_whoami                                 |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67396#019779bc-de1a-4788-81df-a10e2c62b5ef | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest                               | test_init                                         |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67396#019779bc-de1a-4524-873a-63715fc9090a | FAIL          | 0/21     | The test has failed across all retries |
| src/v/cloud_topics/throttler/tests/throttler_test | src/v/cloud_topics/throttler/tests/throttler_test |                  | unit        | https://buildkite.com/redpanda/redpanda/builds/67396#01977981-2b4b-427a-8404-e55fcc02a93c | FAIL          | 0/1      |                                        |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67473">build#67473</a></summary>


| test_class               | test_method                | test_arguments                                                                                                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:-------------------------|:---------------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| CompactionGapsTest       | test_translation_no_gaps   | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1}                                                                                 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977e90-d170-4ee4-a92d-2e9f38ffaebd | FLAKY         | 20/21    | upstream reliability is '92.69662921348315'. current run reliability is '95.23809523809523'. drift is -2.54147 and the allowed drift is set to 50. The test should PASS |
| RandomNodeOperationsTest | test_node_operations       | {"cloud_storage_type": 2, "compaction_mode": "sliding_window", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977e90-d171-45ea-9d22-0c075040956a | FLAKY         | 16/21    | upstream reliability is '100.0'. current run reliability is '76.19047619047619'. drift is 23.80952 and the allowed drift is set to 50. The test should PASS             |
| RedpandaOIDCTlsTest      | test_admin_invalidate_keys |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977e90-d170-4ee4-a92d-2e9f38ffaebd | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest      | test_admin_invalidate_keys |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977eec-49e7-484a-bdbc-2cf09f8ac13d | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest      | test_admin_revoke          |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977e90-d171-45ea-9d22-0c075040956a | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest      | test_admin_revoke          |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977eec-49e4-4c9d-8336-50952ab09e53 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest      | test_admin_whoami          |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977e90-d172-4e23-a987-24784e79b5e4 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest      | test_admin_whoami          |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977eec-49e5-4697-b955-97eb90edb48e | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest      | test_init                  |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977e90-d16f-48b7-9198-1239e08e5eb4 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest      | test_init                  |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67473#01977eec-49e6-4965-9580-8059222feaaa | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67510">build#67510</a></summary>


| test_class          | test_method       | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 |
|:--------------------|:------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|
| RedpandaOIDCTlsTest | test_admin_revoke |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67510#01978004-0d9c-4cb7-892d-ad6a1bdf0227 | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest | test_admin_whoami |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67510#01978004-0d9d-4652-b934-2574a37d820e | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest | test_init         |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67510#01978004-0d9e-4dea-a197-66ee789e4b94 | FAIL          | 0/21     | The test has failed across all retries |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67544">build#67544</a></summary>


| test_class          | test_method       | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 |
|:--------------------|:------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|
| RedpandaOIDCTlsTest | test_admin_revoke |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67544#019782ac-02f4-49ab-81bd-8527e36ba5c3 | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest | test_admin_revoke |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67544#019782af-e305-4c19-942a-1531c728363d | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest | test_admin_whoami |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67544#019782ac-02f5-4314-a9e8-3d80e0e9c7e9 | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest | test_admin_whoami |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67544#019782af-e306-4cb8-b27c-6e6d7bb75557 | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest | test_init         |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67544#019782ac-02f7-4fa4-94cc-70a290585e88 | FAIL          | 0/21     | The test has failed across all retries |
| RedpandaOIDCTlsTest | test_init         |                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67544#019782af-e308-4757-b1d4-56fe18efe45f | FAIL          | 0/21     | The test has failed across all retries |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67578">build#67578</a></summary>


| test_class                                          | test_method                                         | test_arguments                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:----------------------------------------------------|:----------------------------------------------------|:-------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DatalakeDelayedEnablementTest                       | test_enabling_iceberg_in_existing_cluster           | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67578#019784c1-4788-4782-b418-f300ab2c24fb | FLAKY         | 15/21    | upstream reliability is '81.61764705882352'. current run reliability is '71.42857142857143'. drift is 10.18908 and the allowed drift is set to 50. The test should PASS |
| RestCatalogConnectionTest                           | test_redpanda_connection_to_rest_catalog            | {"cloud_storage_type": 1}                              | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67578#0197851a-db28-4219-8092-fbecb25d865e | FLAKY         | 19/21    | upstream reliability is '96.5'. current run reliability is '90.47619047619048'. drift is 6.02381 and the allowed drift is set to 50. The test should PASS               |
| RaftAvailabilityTest                                | test_controller_node_isolation                      |                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67578#019784c1-4788-4534-b86c-fe8c2d06d251 | FLAKY         | 20/21    | upstream reliability is '94.5578231292517'. current run reliability is '95.23809523809523'. drift is -0.68027 and the allowed drift is set to 50. The test should PASS  |
| src/v/kafka/server/tests/quota_manager_rpbench_test | src/v/kafka/server/tests/quota_manager_rpbench_test |                                                        | unit        | https://buildkite.com/redpanda/redpanda/builds/67578#0197846c-7b11-46be-978f-9619f6ef86f7 | FAIL          | 0/1      |                                                                                                                                                                         |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67633">build#67633</a></summary>


| test_class                 | test_method                                       | test_arguments                                                                                                                          | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:---------------------------|:--------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| PartitionReassignmentsTest | test_add_partitions_with_inprogress_reassignments |                                                                                                                                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67633#01978a5b-c8ff-463f-8c9e-c5b36a68eb95 | FLAKY         | 17/21    | upstream reliability is '93.29173166926678'. current run reliability is '80.95238095238095'. drift is 12.33935 and the allowed drift is set to 50. The test should PASS |
| RandomNodeOperationsTest   | test_node_operations                              | {"cloud_storage_type": 1, "compaction_mode": "sliding_window", "enable_failures": true, "mixed_versions": false, "with_iceberg": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67633#01978a5b-c900-4463-bd8b-ccf377a8468c | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS              |
| RedpandaOIDCTlsTest        | test_admin_invalidate_keys                        |                                                                                                                                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67633#01978a5b-c8fe-46e6-8dca-71480f8bd17a | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest        | test_admin_revoke                                 |                                                                                                                                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67633#01978a5b-c8ff-463f-8c9e-c5b36a68eb95 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest        | test_admin_whoami                                 |                                                                                                                                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67633#01978a5b-c900-4463-bd8b-ccf377a8468c | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| RedpandaOIDCTlsTest        | test_init                                         |                                                                                                                                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67633#01978a5b-c900-43ab-9607-89df0cd33ee1 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
</details>
<!--/body-->

rockwotj commented:
/ci-repeat

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/67380-->
<!--c9ed4e78-5906-4035-b473-e4b0a1e4db45-->
<!--b3b24e66-368a-49a2-834e-4bfb623e9c60-->
<!--eb7a0c8f-c790-490c-9cf5-816ad2685abe-->
<!--38dccd4b-3c8f-4870-90c8-c064c46ab641-->
<!--26d0a2cd-8b30-402d-b204-4014de7c16c9-->
<!--48b9e22a-9733-4bea-9817-90bd942f2842-->
<!--752d7245-85c6-402a-b285-915a5a9845de-->
<!--5901eee2-abe9-44db-94bb-b97dd4d28f8d-->
<!--title-->
## Retry command for [Build#67380](https://buildkite.com/redpanda/redpanda/builds/67380)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_invalidate_keys
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init
```
<!--/body-->

Lazin commented:
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_invalidate_keys
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/67396-->
<!--d789f7ef-99a2-489a-a978-e7e5d12a37a6-->
<!--322f8c95-bba5-4f22-9ec6-9040050de48b-->
<!--0a157656-6b4c-4bc3-bf85-01e552fd10bb-->
<!--2f47768c-ea1f-4971-8913-b4d3f3529f0c-->
<!--title-->
## Retry command for [Build#67396](https://buildkite.com/redpanda/redpanda/builds/67396)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_invalidate_keys
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
```
<!--/body-->

Lazin committed:
http: Fix 'with_client' template

The function invokes 'stop' method of the http client and then
'shutdown'. There is no need to invoke 'shutdown' after 'stop' since
'stop' is shutting down the TCP connection gracefully.

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/67473-->
<!--c935cb7e-7bee-4287-99e0-e849a63173e3-->
<!--b292a8f8-2b90-4018-9a18-e7cde83c9d6a-->
<!--c1a9a798-25b9-4819-b831-e221080fbc13-->
<!--015e4ac5-6228-44fc-a729-d9f523293d75-->
<!--6ea8fdbb-7b4e-4b44-bbc8-35cda890e919-->
<!--82b25513-3438-402b-ba75-6df4f3211ae9-->
<!--title-->
## Retry command for [Build#67473](https://buildkite.com/redpanda/redpanda/builds/67473)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
```
<!--/body-->

Lazin commented:
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/67510-->
<!--2183f0ea-3317-4676-b75a-856381a5b0be-->
<!--e6988d43-c52a-448b-92b9-4f8f39bcf270-->
<!--80cb76a6-9b64-4dc2-92de-06e2d41914e8-->
<!--title-->
## Retry command for [Build#67510](https://buildkite.com/redpanda/redpanda/builds/67510)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init
```
<!--/body-->

Lazin commented:
a bunch of tests are failing with `failed to start within 40 seconds` error, restarting failed tests

Lazin commented:
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/67544-->
<!--fb3d775d-dab0-47cd-a422-44a3b37b390f-->
<!--ba4e3167-f1d3-45fe-a930-9d5f60fe5d8e-->
<!--b8667e47-6327-4a4a-b89d-0bf33d2bf80a-->
<!--7da733fa-13e2-407c-82b9-1dec18295c60-->
<!--a2788180-67c9-4fab-a8a8-7d788841e8c3-->
<!--a08ca089-cb3b-4c0f-a687-18058110d46e-->
<!--title-->
## Retry command for [Build#67544](https://buildkite.com/redpanda/redpanda/builds/67544)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init
```
<!--/body-->

rockwotj commented on src/v/http/client.h:
```diff
@@ -330,12 +330,8 @@ auto with_client(client&& cl, Func func) {
       std::move(cl), [func = std::move(func)](client& cl) mutable {
-          return ss::futurize_invoke(func, cl).finally([&cl] {
-              return cl.stop().then([&cl] {
-                  cl.shutdown();
```
Are there other usages like this in the code base?

For example is this a problem:

https://github.com/redpanda-data/redpanda/blob/9ca56e0141c65171dddc90cff6adc2eb5e5e9728/src/v/cloud_storage_clients/abs_client.cc#L466-L467

Or is that just a noop? Should we update that code to only call stop? I'm assuming it's not a problem since that code is exercised a lot in ducktape, but maybe it's worth updating the comments on `wait_input_shutdown` on when it can/should be called.

there do seem to be additional opportunities for cleanup we should chase (to prevent cargo culting the wrong pattern).

Lazin replied:
that makes sense, the problem is that if this method is called in stop automatically the oauth tests starts to fail
I'll try to chase this

Lazin committed:
http: Invoke wait_input_shutdown in stop

and avoid calling it manually, the stop can only be completed when the
underlying client disconnects and it's safe to destroy the http client.

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

rockwotj commented on src/v/http/client.cc:
```diff
@@ -266,6 +266,7 @@ ss::future<> client::stop() {
     _stopped = true;
     co_await _connect_gate.close();
     shutdown();
+    co_await base_transport::wait_input_shutdown();
```
I believe this is only used in `abs_client` and it's called *after* stop...

Lazin replied:
because 'stop' invokes 'shutdown' in old code and this method waits until the peer disconnects

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/67633-->
<!--5cc04421-3a0a-4c19-8b6a-0dc02bc68e93-->
<!--cc804aa8-5b6d-4795-ac82-a6f35c43a591-->
<!--64aa6040-6249-42d2-aacb-ebdd7f10395a-->
<!--6da783ff-3184-427d-9e93-55007085d539-->
<!--title-->
## Retry command for [Build#67633](https://buildkite.com/redpanda/redpanda/builds/67633)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_init
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_invalidate_keys
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_revoke
tests/rptest/tests/redpanda_oauth_test.py::RedpandaOIDCTlsTest.test_admin_whoami
```
<!--/body-->
